src/main/kotlin/org/jetbrains/research/transformations/anonimization/Anonymization.kt

```diff
@@ -0,0 +1,164 @@
+@file:Suppress("UNCHECKED_CAST")
+
+package org.jetbrains.research.transformations.anonimization
+
+import com.github.gumtreediff.tree.ITree
+import com.github.gumtreediff.tree.TreeContext
+import org.jetbrains.research.transformations.NodeType
+import org.jetbrains.research.transformations.Transformation
+import org.jetbrains.research.transformations.nodes
+
+object Anonymization : Transformation {
+    override val metadataKey = "anonymization"
+    private const val VAR_PREFIX = "v"
+    private const val FUN_PREFIX = "f"
+    private const val ARG_PREFIX = "a"
+    private const val EMPTY_PREFIX = ""
+    private const val SEPARATOR = "_"
+
+    // Check if the node has the parent function and get its name
+    // Otherwise return EMPTY_PREFIX
+    private fun getNodeLabelPrefix(node: ITree, oldPrefix: String,
+                                   metaInformation: AnonymizationMetaInformation): String {
+        var prefix = EMPTY_PREFIX
+        val parentFunctions = node.parents.map { it.label }.filter { it.isNotEmpty() }
+        if (parentFunctions.isNotEmpty()) {
+            prefix = parentFunctions[0] + SEPARATOR
+        }
+        if (oldPrefix != prefix) {
```

Comment:
Why should we reset ids here?

Reply:
because I need to reset ids for each new function, as Nastya told me:
```
def f1 (f1_a1):
   f1_v1 = f1_a1 + 1
   return f1_v1

def f2 (f2_a1):
   f2_v1 = f2_a1 + 2
   return f2_v1
```